pkg/providers/vsphere/defaults.go

```diff
@@ -16,14 +16,13 @@ type defaulter struct {
 	govc ProviderGovcClient
 }
 
-func newDefaulter(govc ProviderGovcClient) *defaulter {
+func NewDefaulter(govc ProviderGovcClient) *defaulter {
 	return &defaulter{
 		govc: govc,
 	}
 }
 
-func (d *defaulter) setDefaults(ctx context.Context, spec *spec) error {
```

Comment:
we could maintain `setDefaults` and call `setDefaultsForMachineConfig` and `SetDefaultsForDatacenterConfig` from it
to keep the provider "simpler" (lmao)

Reply:
the reason i removed the setDefaults is in `SetupAndValidateCreateCluster` method, we need to follow certain order: 1. SetDefaultsForDatacenterConfig 2. ValidateVCenterConfig 3. setDefaultsForMachineConfig. the 3rd step only makes sense after vCenterConfig is valid